src/components/SubNav/SubNav.tsx

```diff
@@ -0,0 +1,312 @@
+  isSelected,
+  screenreaderOnlyText = "",
+}) => {
+  const childrenStyles = useMultiStyleConfig("SubNavChildren", {
+    isOutlined: isOutlined,
+  });
+
+  return (
+    <li>
+      <Button
+        id={id}
+        buttonType="text"
+        className={isSelected ? "selectedItem" : ""}
+        sx={{ ...childrenStyles.outLine }}
+        screenreaderOnlyText={screenreaderOnlyText}
+      >
+        {children}
+      </Button>
+    </li>
+  );
+};
+
+export const SubNavLink: React.FC<React.PropsWithChildren<any>> = ({
+  id,
+  children,
+  isOutlined,
+  isSelected,
+  href,
+  screenreaderOnlyText = "",
+}) => {
+  const childrenStyles = useMultiStyleConfig("SubNavChildren", {
+    isOutlined: isOutlined,
+  });
+
+  return (
+    <li>
+      <Link
+        key={id}
+        id={id}
+        type="action"
+        href={href}
+        isUnderlined={false}
+        screenreaderOnlyText={screenreaderOnlyText}
+        className={isSelected ? "selectedItem" : ""}
+        sx={{ ...childrenStyles.outLine }}
+      >
+        {children}
+      </Link>
+    </li>
+  );
+};
+
+/**
+ * The `SubNav` component is a navigation element that displays a group of
+ * related action items (buttons or links) in a horizontal layout.  The action
+ * items in the `SubNav` component will link to children within or perform
+ * actions related to the current section of a website.
+ */
+export const SubNav: ChakraComponent<
+  React.ForwardRefExoticComponent<
+    React.PropsWithChildren<SubNavProps> & React.RefAttributes<HTMLDivElement>
+  >,
+  SubNavProps
+> = chakra(
+  forwardRef<HTMLDivElement, React.PropsWithChildren<SubNavProps>>(
+    (props, _ref?) => {
+      const {
+        className,
+        actionBackgroundColor,
+        highlightColor,
+        primaryActions,
+        secondaryActions,
+      } = props;
+
+      const [showRightFade, setShowRightFade] = useState(false);
+      const [lastScrollLeft, setLastScrollLeft] = useState(0);
+      const scrollableRef = useRef(null);
+
+      if (actionBackgroundColor !== undefined && highlightColor === undefined) {
+        console.warn(
+          "NYPL Reservoir SubNav: The `actionBackgroundColor` prop has been passed, but the `highlightColor` prop has not been passed. Because of this, the `actionBackgroundColor` prop will be ignored."
+        );
+      }
+
+      const validateActions = (actions: React.ReactNode, propName: string) => {
+        if (React.isValidElement(actions)) {
+          React.Children.forEach(
+            actions.props.children,
+            (child: React.ReactElement) => {
+              if (child.type !== SubNavButton && child.type !== SubNavLink) {
+                console.warn(
+                  `NYPL Reservoir SubNav: An element that is not a SubNavButton or SubNavLink component has been passed in the \`${propName}\` prop. That element will be ignored.`
+                );
+                return null;
+              }
+            }
+          );
+        }
```

Comment:
In theory this is good but this only logs the warning. This does not actually ignore the element. In the screenshot I passed a DS `Button` and an HTML anchor link:

<img width="547" alt="Screenshot 2024-11-13 at 7 21 17 PM" src="https://github.com/user-attachments/assets/f6c6c441-e5f2-4923-915a-ef4ee80baaa7">


If we actually want to ignore the element, we'll have to do more work which I think we can not do for this round. What do you think about this being the responsibility of the consuming application? @bigfishdesign13  @oliviawongnyc ?

Reply:
We do a similar type check of child components in the `ButtonGroup` and `List` components, so yes, I think we can continue that pattern here. But let's only throw a warning instead of ignoring the elements.

@deepikagonuguntla Update the warning to the following:
```
"NYPL Reservoir SubNav: An element that is not a 
`SubNavButton` or `SubNavLink` component has 
been passed in the \`${propName}\` prop. That 
element may not work properly."
```